src/main/groovy/jenkins/automation/utils/CheckmarxUtils.groovy

```diff
@@ -0,0 +1,172 @@
+            !**/obj/**/*,  !**/backup/**/*, !**/.idea/**/*, !**/*.DS_Store, !**/*.ipr,     !**/*.iws,
+            !**/*.bak,     !**/*.tmp,       !**/*.aac,      !**/*.aif,      !**/*.iff,     !**/*.m3u, !**/*.mid, !**/*.mp3,
+            !**/*.mpa,     !**/*.ra,        !**/*.wav,      !**/*.wma,      !**/*.3g2,     !**/*.3gp, !**/*.asf, !**/*.asx,
+            !**/*.avi,     !**/*.flv,       !**/*.mov,      !**/*.mp4,      !**/*.mpg,     !**/*.rm,  !**/*.swf, !**/*.vob,
+            !**/*.wmv,     !**/*.bmp,       !**/*.gif,      !**/*.jpg,      !**/*.png,     !**/*.psd, !**/*.tif, !**/*.swf,
+            !**/*.jar,     !**/*.zip,       !**/*.rar,      !**/*.exe,      !**/*.dll,     !**/*.pdb, !**/*.7z,  !**/*.gz,
+            !**/*.tar.gz,  !**/*.tar,       !**/*.gz,       !**/*.ahtm,     !**/*.ahtml,   !**/*.fhtml, !**/*.hdm,
+            !**/*.hdml,    !**/*.hsql,      !**/*.ht,       !**/*.hta,      !**/*.htc,     !**/*.htd, !**/*.war, !**/*.ear,
+            !**/*.htmls,   !**/*.ihtml,     !**/*.mht,      !**/*.mhtm,     !**/*.mhtml,   !**/*.ssi, !**/*.stm,
+            !**/*.stml,    !**/*.ttml,      !**/*.txn,      !**/*.xhtm,     !**/*.xhtml,   !**/*.class, !**/*.iml, !Checkmarx/Reports/*.*
+        """.stripIndent()
+
+        context.with {
+            configure {
+                it / builders / 'com.checkmarx.jenkins.CxScanBuilder' {
+
+                    'useOwnServerCredentials'(
+                        cxConfig.get(
+                            "useOwnServerCredentials",
+                            useOwnServerCredentialsDefault
+                        )
+                    )
+
+                    if (cxConfig.useOwnServerCredentials) {
+                        'serverUrl'(cxConfig.serverUrl)
+                        'username'(cxConfig.username)
+                        'password'(cxConfig.password)
+                    }
+
+                    'projectName'(cxConfig.projectName)
+                    'groupId'(cxConfig.groupId) // really the Checkmarx 'team'
+                    'preset'(cxConfig.get("preset", presetDefault))
+
+                    // TODO figure out whether this does anything
+                    'presetSpecified'(false)
+
+                    // TODO probably this should be configurable
+                    'exclusionsSetting'('job')
+
+                    'excludeFolders'(
+                        cxConfig.get("excludeFolders", excludeFoldersDefault)
+                    )
+                    'filterPattern'(
+                        cxConfig.get("filterPattern", filterPatternDefault)
+                    )
+
+                    'incremental'(cxConfig.get("incremental", incrementalDefault))
+
+                    // TODO is this right?
+                    'fullScansScheduled'(false)
+                    'fullScanCycle'(
+                        cxConfig.get("fullScanCycle", fullScanCycleDefault)
+                    )
+
+                    // TODO should we really be setting this?
```

Comment:
I thought group consensus was that incremental scans were faster. Don't we want to be able to do this?

Reply:
Definitely group consensus on that, and we are defaulting to incremental, see line 67. This is confusing: for some reason the Cx plugin creates this element and sets it to `false`, even when you have incremental enabled. I don't know why -- nor do I know what it does. Hence my comment. If you have any insights, lmk.

Reply:
To be clear: this seems to have no impact on whether the scans are incremental, and we ARE defaulting to incremental, as best I can tell. But this was carried over from the older builder, and it is an element that's created (and set to `false`) when creating a Cx job via the Jenkins UI. No idea.